### DISCUSSION TITLE
ButtonMessage message appears at desktop browser, but not appearing at mobile device

### DISCUSSION THREAD

**Discussion by rvp98** [tulir/whatsmeow#226] [WhatsApp protocol Q&A] [answered]
```
msg := &waProto.Message{ButtonsMessage: &waProto.ButtonsMessage{
							HeaderType:  waProto.ButtonsMessage_EMPTY.Enum(),
							ContentText: proto.String("Hello"),
							FooterText:  proto.String("Hi"),
							Buttons: []*waProto.Button{
								{
									ButtonId: proto.String("id1"),
									ButtonText: &waProto.ButtonText{
										DisplayText: proto.String("HALO"),
									},
									Type: waProto.Button_RESPONSE.Enum(),
								},
								{
									ButtonId: proto.String("id2"),
									ButtonText: &waProto.ButtonText{
										DisplayText: proto.String("HI"),
									},
									Type: waProto.Button_RESPONSE.Enum(),
								},
								{
									ButtonId: proto.String("id3"),
									ButtonText: &waProto.ButtonText{
										DisplayText: proto.String("ALO"),
									},
									Type: waProto.Button_RESPONSE.Enum(),
								},
							},
						}}
```

![image](https://user-images.githubusercontent.com/54128956/183566898-a893cdb6-e7a6-4218-9c6f-55c1176b5095.png)
![image](https://user-images.githubusercontent.com/54128956/183567083-96332cfb-6dce-4d24-920f-80d13566c12f.png)

Button message appear on desktop version of whatsaapp but not appear on mobile device

**Comment by Diden05** [comment, marked as answer]
see https://github.com/tulir/whatsmeow/issues/222#issuecomment-1206060240

**Reply by cakwahyu** [reply]
I experienced the same case, on the sender's page it appeared (using iOS), but on the recipient's page (iOS) it didn't appear

**Comment by rvp98** [comment]
Still not working i already add the **`Header`**

```
msg := &waProto.Message{
     ButtonsMessage: &waProto.ButtonsMessage{
        HeaderType: waProto.ButtonsMessage_TEXT.Enum(),
	Header: &waProto.ButtonsMessage_Text{
	      Text: "Title",
	},
	ContentText: proto.String(rc.Respond),
	FooterText:  proto.String("Gunakan tombol di bawah ini untuk memilih keyword yang anda butuhkan"),
	Buttons: buttonList,
	},
}
```

**Comment by mamur-rezeki** [comment]
Make sure every type of your Buttons is correct,

For common button response type :
```go
waProto.ButtonsMessage_Button{
     ButtonId: proto.String("id1"),
     ButtonText: &waProto.ButtonsMessage_Button_ButtonText{
       DisplayText: proto.String("btn1"),
     },

     // If we set the type as another type, 
     // maybe another field(s) should be filled also.
     // e : NativeFlowInfo for ... _NATIVE_FLOW type.
     Type: waProto.ButtonsMessage_Button_RESPONSE.Enum(), 
}
```

**Reply by mamur-rezeki** [reply]
I've been realized on your code : 

```go
...
    Buttons: []*waProto.Button{},
...
```
It should be ` []*waProto.ButtonsMessage_Button{} `

**Reply by rvp98** [reply]
Thanks for the answer, Wait let me try it

**Comment by hrizal** [comment]
header and footer have maximal character, if reach max char, will not work

Pada tanggal Rab, 10 Agu 2022 09.10, rvp98 ***@***.***>
menulis:

**Reply by mamur-rezeki** [reply]
I didn't know that's true, but I've been tested with ` 3405 ` length of ` string ` haven't noticed any error for limitation.

Btw, I'm using Android & web version, both have no problem.

![Screenshot_20220810-152354_WhatsApp](https://user-images.githubusercontent.com/107448645/183840303-531339b7-7923-4cdc-b1f7-cc0980935e36.jpg)

**Comment by hrizal** [comment]
![image](https://user-images.githubusercontent.com/47385704/184592791-c0221172-acbc-40c1-a74d-258d1f3598ac.png)

it is limitation if using official api

**Reply by rvp98** [reply]
Thanks for make detailing description for the limitation
